fl4health/clients/basic_client.py

```diff
@@ -273,6 +273,20 @@ def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, Dict
         # Update after train round (Used by Scaffold and DP-Scaffold Client to update control variates)
         self.update_after_train(local_steps, loss_dict)
 
+        # Check if we should run an evaluation with validation data after fit (used by FedDGGA)
+        try:
+            evaluate_after_fit = self.narrow_config_type(config, "evaluate_after_fit", bool)
+        except ValueError:
+            evaluate_after_fit = False
+
+        if evaluate_after_fit:
+            loss, metric_values = self.validate()
```

Comment:
perhaps we can pull this, and the below, into a function called `evaluated_after_fit` or something like it. That way, participants could potentially override it to get different behavior as well?

Reply:
Done, please check.